compat/os/mingw.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_OS_WINDOWS_MINGW_H
+#define SQUID_OS_WINDOWS_MINGW_H
+
+#if _SQUID_MINGW_
```

Comment:
Code in libcompat should (only) be added after build and testing using the environment+toolchain identifies a real need. The followup PRs in this project will be adding the portability hacks back as and when necessary based on fresh builds @kinkie and I are doing.

Reply:
Squid did not build with MinGW-w64 before this, and does not afterward either.

What we are getting here is a clean way to avoid the complexity and side effects @kinkie was faced with in PR #1085.